File: jfingerprintmanager/src/test/java/com/jesusm/jfingerprintmanager/encryption/presenter/FingerprintEncryptionDialogPresenterTest.java
package com.jesusm.jfingerprintmanager.encryption.presenter;

import com.jesusm.jfingerprintmanager.base.BasePresenterTest;
import com.jesusm.jfingerprintmanager.authentication.presenter.FingerprintAuthenticationDialogPresenter;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.runners.MockitoJUnitRunner;

import static org.mockito.Mockito.verify;

@RunWith(MockitoJUnitRunner.class)
public class FingerprintEncryptionDialogPresenterTest extends BasePresenterTest {

    @Test
    public void shouldCloseIfFingerprintNotAvailable() {
        FingerprintAuthenticationDialogPresenter presenter = createPresenter(mockView);
        presenter.onViewShown();

        verify(mockView).onFingerprintDisplayed();
    }
}

File: jfingerprintmanager/src/test/kotlin/com.jesusm.jfingerprintmanager/encryption/presenter/FingerprintEncryptionDialogPresenterTest.kt
package com.jesusm.jfingerprintmanager.encryption.presenter

import com.jesusm.jfingerprintmanager.base.BasePresenterTest
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mockito.verify
import org.mockito.runners.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class FingerprintEncryptionDialogPresenterTest : BasePresenterTest(){
    @Test
    fun shouldCloseIfFingerprintNotAvailable() {
        val presenter = createPresenter(mockView)
        presenter.onViewShown()

        verify(mockView).onFingerprintDisplayed()
    }
}